Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp

```diff
@@ -1252,6 +1281,58 @@ void ezDebugRenderer::RenderInternal(const ezDebugRendererContext& context, cons
     }
   }
 
+  // info text
```

Comment:
You have to do this whole code block before the framecounter is set to the current frame:
pDoubleBufferedContextData->m_uiLastRenderedFrame = ezRenderWorld::GetFrameCounter();
otherwise you introduce a frame delay when you call Draw2DText here

Reply:
I have changed this, but I sometimes see flickering, ie the text is rendered twice in a single frame. I'm wondering, whether that's because of my change here, or whether the text was always potentially inserted twice (mesh asset stats rendering).

Reply:
Actually I was wrong: The code block needs to be before
PerContextData* pData = pDoubleBufferedContextData->m_pData[ezRenderWorld::GetDataIndexForRendering()].Borrow();
That should solve the flickering.

Reply:
Yes, and since I need pData, I also needed to use GetDataIndexForExtraction actually.

Reply:
Can you add a unit test where you print a framecounter with Draw2DText and DrawInfoText and check that the frame counter is the same.

Reply:
So I just noticed that using the IndexForRendering, some strings (e.g. the FPS counter) don't show up anymore. So I switched it back to the other index. But then I sometimes get flickering.

I don't understand the problem to be honest.

Reply:
IndexForRendering should be correct, but I can take a look later

Reply:
I've added a unit test. Not sure whether it makes sense though.

Reply:
Is it possible that you tried to enable the fps counter in editor? That was not working because it only searched for a main view but not an editor view as fallback, I submitted a fix. Now everything seems to work as expected as far as I can tell.

Reply:
No, that's not the problem, but interesting that you made that work.

The flickering is still there (open a mesh asset and transform something, then you'll see the text flicker). I guess I know what the problem is, but it might not be (easily) fixable. For now it's not that important and we can talk about that at a later time.